bundles/org.openhab.binding.govee/src/main/java/org/openhab/binding/govee/internal/GoveeHandler.java

```diff
@@ -0,0 +1,394 @@
+                            logger.debug("Color Temperature Absolute change with Percent Type {}", command.toString());
+                            lastColorTempInKelvin = quantity.intValue();
+                            logger.debug("COLOR_TEMPERATURE_ABS: lastColorTempInKelvin {}", lastColorTempInKelvin);
+                            int lastColorTempInPercent = ((Double) ((lastColorTempInKelvin
+                                    - COLOR_TEMPERATURE_MIN_VALUE)
+                                    / (COLOR_TEMPERATURE_MAX_VALUE - COLOR_TEMPERATURE_MIN_VALUE) * 100.0)).intValue();
+                            logger.debug("computed lastColorTempInPercent {}", lastColorTempInPercent);
+                            sendColorTemp(lastColorTempInKelvin);
+                        }
+                        break;
+                }
+            }
+            if (!thing.getStatus().equals(ThingStatus.ONLINE)) {
+                updateStatus(ThingStatus.ONLINE);
+            }
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "@text/offline.communication-error.could-not-query-device [\"" + goveeConfiguration.hostname
+                            + "\"]");
+        }
+    }
+
+    /**
+     * Initiate a refresh to our thing devicee
+     *
+     */
+    private void triggerDeviceStatusRefresh() throws IOException {
+        if (refreshRunning) {
+            return;
+        }
+        if (GoveeDiscoveryService.isDiscoveryActive()) {
+            logger.debug("Not triggering refresh as Scan is currently active");
+            return;
+        }
+        refreshRunning = true;
+
+        logger.debug("trigger Refresh Status of device {}", thing.getLabel());
+
+        try {
+            GenericGoveeRequest lightQuery = new GenericGoveeRequest(
+                    new GenericGoveeMsg("devStatus", new EmptyValueQueryStatusData()));
+            send(GSON.toJson(lightQuery));
+        } finally {
+            refreshRunning = false;
+        }
+    }
+
+    public void sendColor(Color color) throws IOException {
+        lastColor = color;
+        GenericGoveeRequest lightColor = new GenericGoveeRequest(
+                new GenericGoveeMsg("colorwc", new ColorData(color, 0)));
+        send(GSON.toJson(lightColor));
+    }
+
+    public void sendBrightness(int brightness) throws IOException {
+        lastBrightness = brightness;
+        GenericGoveeRequest lightBrightness = new GenericGoveeRequest(
+                new GenericGoveeMsg("brightness", new ValueIntData(brightness)));
+        send(GSON.toJson(lightBrightness));
+    }
+
+    private void sendOnOff(OnOffType switchValue) throws IOException {
+        lastOnOff = (switchValue == OnOffType.ON) ? 1 : 0;
+        GenericGoveeRequest switchLight = new GenericGoveeRequest(
+                new GenericGoveeMsg("turn", new ValueIntData(lastOnOff)));
```

Comment:
What will happen if device is already off with brightness = 0? Will it be turned on with some brightness value, or will it be turned on with brightness = 0, so that it actually doesn't emit light?

Reply:
I always know during regularly updates that I receive from the device what the devices knows about the brightness. If I then switch on the device via openHAB it uses that brightness. Believe I even did almost an hour on testing different cross scenarios. I works as it supposed to work.